hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieClusteringJob.java

```diff
@@ -171,4 +200,38 @@ private int doCluster(JavaSparkContext jsc) throws Exception {
       return client.scheduleClustering(Option.empty());
     }
   }
+
+  @TestOnly
+  public int doScheduleAndCluster() throws Exception {
+    return this.doScheduleAndCluster(jsc);
+  }
+
+  public int doScheduleAndCluster(JavaSparkContext jsc) throws Exception {
+    LOG.info("Step 1: Do schedule");
+    String schemaStr = getSchemaFromLatestInstant();
+    try (SparkRDDWriteClient client = UtilHelpers.createHoodieClient(jsc, cfg.basePath, schemaStr, cfg.parallelism, Option.empty(), props)) {
+
+      Option<String> instantTime;
+      if (cfg.clusteringInstantTime != null) {
+        client.scheduleClusteringAtInstant(cfg.clusteringInstantTime, Option.empty());
+        instantTime = Option.of(cfg.clusteringInstantTime);
+      } else {
+        instantTime = client.scheduleClustering(Option.empty());
+      }
+
+      int result = instantTime.isPresent() ? 0 : -1;
```

Comment:
1、can we implement three function?
 doschedule、docluster、doScheduleAndCluster 
2、doScheduleAndCluster is consists of doschedule and docluster
doScheduleAndCluster {
doschedule()
docluster()
}

Reply:
Emmmm, actually, there already has doSchedule() and doCluster() function. But if we let doScheduleAndCluster() use  doschedule() and docluster() directly, it will start and stop SparkRDDWriteClient twice which is an expensive action and unnecessary. 

Maybe let schedule action and cluster action use a common SparkRDDWriteClient is better.

For example start and stop Timeline service twice.
```
21/07/15 11:05:11 INFO EmbeddedTimelineService: Starting Timeline service !!
21/07/15 11:05:11 INFO EmbeddedTimelineService: Overriding hostIp to (localhost) found in spark-conf. It was null
21/07/15 11:05:11 INFO FileSystemViewManager: Creating View Manager with storage type :MEMORY
21/07/15 11:05:11 INFO FileSystemViewManager: Creating in-memory based Table View
21/07/15 11:05:11 INFO log: Logging initialized @4500ms to org.apache.hudi.org.eclipse.jetty.util.log.Slf4jLog
21/07/15 11:05:11 INFO Javalin: 
           __                      __ _
          / /____ _ _   __ ____ _ / /(_)____
     __  / // __ `/| | / // __ `// // // __ \
    / /_/ // /_/ / | |/ // /_/ // // // / / /
    \____/ \__,_/  |___/ \__,_//_//_//_/ /_/

        https://javalin.io/documentation

21/07/15 11:05:11 INFO Javalin: Starting Javalin ...
```

Reply:
can we reuse the sparkclient
modify doCluster(sc) to doCluster(sc, client)

Reply:
Nice idea. Changed. PTAL :)